src/QtLocationPlugin/QMLControl/QGCMapEngineManager.cc

```diff
@@ -176,196 +155,168 @@ QGCMapEngineManager::startDownload(const QString& name, const QString& mapType)
+        QGCCachedTileSet* const set = qobject_cast<QGCCachedTileSet*>(_tileSets->get(i));
+        if (set && (set->id() == setID)) {
+            _tileSets->removeAt(i);
+            delete set;
+            emit tileSetsChanged();
+            return;
         }
     }
-    if(setToDelete) {
-        _tileSets.removeAt(i);
-        delete setToDelete;
-    }
-    emit tileSetsChanged();
 }
 
-//-----------------------------------------------------------------------------
-void
-QGCMapEngineManager::taskError(QGCMapTask::TaskType type, QString error)
+void QGCMapEngineManager::taskError(QGCMapTask::TaskType type, const QString &error)
 {
     QString task;
-    switch(type) {
+    switch (type) {
     case QGCMapTask::taskFetchTileSets:
-        task = "Fetch Tile Set";
+        task = QStringLiteral("Fetch Tile Set");
         break;
     case QGCMapTask::taskCreateTileSet:
-        task = "Create Tile Set";
+        task = QStringLiteral("Create Tile Set");
         break;
     case QGCMapTask::taskGetTileDownloadList:
-        task = "Get Tile Download List";
+        task = QStringLiteral("Get Tile Download List");
         break;
     case QGCMapTask::taskUpdateTileDownloadState:
-        task = "Update Tile Download Status";
+        task = QStringLiteral("Update Tile Download Status");
         break;
     case QGCMapTask::taskDeleteTileSet:
-        task = "Delete Tile Set";
+        task = QStringLiteral("Delete Tile Set");
         break;
     case QGCMapTask::taskReset:
-        task = "Reset Tile Sets";
+        task = QStringLiteral("Reset Tile Sets");
         break;
     case QGCMapTask::taskExport:
-        task = "Export Tile Sets";
+        task = QStringLiteral("Export Tile Sets");
         break;
     default:
-        task = "Database Error";
+        task = QStringLiteral("Database Error");
         break;
     }
+
     QString serror = "Error in task: " + task;
     serror += "\nError description:\n";
     serror += error;
-    qWarning() << "QGCMapEngineManager::_taskError()";
+
     setErrorMessage(serror);
+
+    qCWarning(QGCMapEngineManagerLog) << serror;
 }
 
-//-----------------------------------------------------------------------------
-void
-QGCMapEngineManager::_updateTotals(quint32 totaltiles, quint64 totalsize, quint32 defaulttiles, quint64 defaultsize)
+void QGCMapEngineManager::_updateDiskFreeSpace()
 {
-    for(int i = 0; i < _tileSets.count(); i++ ) {
-        QGCCachedTileSet* set = qobject_cast<QGCCachedTileSet*>(_tileSets.get(i));
+    const QString path = getQGCMapEngine()->getCachePath();
+    if (path.isEmpty()) {
+        return;
+    }
+
+    const QStorageInfo info(path);
+
+    const quint32 total = static_cast<quint32>(info.bytesTotal() / 1024);
+    _diskSpace = total;
+
+    const quint32 free = static_cast<quint32>(info.bytesFree() / 1024);
+    setFreeDiskSpace(free);
+
+    qCDebug(QGCMapEngineManagerLog) << info.rootPath() << "has" << free << "Mbytes available.";
```

Comment:
The way it's calculated, it's in KB. There was also a comment below saying it's in MB, so maybe the calculation needs to be fixed.
```suggestion
    qCDebug(QGCMapEngineManagerLog) << info.rootPath() << "has" << free << "Kbytes available.";
```

Reply:
DiskSpace/FreeDiskSpace are currently unused, but they could be useful for Don to add in the UI so I just made a note about it